documentation/src/main/asciidoc/topics/ref_memcached_client_interoperability.adoc

```diff
@@ -0,0 +1,21 @@
+[id='memcached_interoperability-{context}']
+= Memcached Clients and Storage Formats
+The {brandname} Server Memcached endpoint is deprecated and disabled by
+default. If you enable the Memcached endpoint, you should use a text-based
+storage format for data.
+
+[IMPORTANT]
+====
+The Memcached endpoint does not support authentication. For security purposes
+you should use dedicated caches for Memcached clients. You should not use REST
+or Hot Rod clients to interact on the same data set as Memcached clients.
+====
+
+Memcached clients can handle keys only as String and values stored as `byte[]`.
+However, Memcached clients cannot handle deserialized objects (POJOs).
+
```

Comment:
```master``` is more detailed about memcached. I've distilled the info here:


To use a memcached client with Infinispan, your cache must use the `text/plain` media type for the keys, and the
value can be any arbitrary ```byte[]```.

-----

Most Java Memcached clients have the capability of plugging a marshaller, so the experience is similar to a Hot Rod client: POJOs are used in the client side, that are converted to/from ```byte[]``` when interacting with the server. 

Less capable memcached clients don't have the support for marshallers, and can only read and write the entry values "as is". 
To better support those clients, it is possible to configure the Memcached server so that it always convert the value to specific format (https://infinispan.org/docs/stable/titles/memcached/memcached.html#memcached_client_encoding)

An example of usage is a cache that stores protobuf, but with a Memcached connector having ```client-encoding``` configured as ```application/json```: memcached clients can read and write json documents, but the storage is a protobuf ```byte[]```

-----

Reply:
thanks @gustavonalle I don't know why I've been faffing about with the Memcached stuff so much. I've had another go at it and have hopefully captured all the right details now. I think adding an example cache encoding config is helpful. please let me know if you've more nits.